### PR TITLE
Bringing Facbebook\URL namespace into PSR2 compliance

### PR DESCRIPTION
**Pull request by stevenmaguire**
It's happening.

### CODE REVIEW

**Comment by SammyK** [comment]
:+1:

File: src/Facebook/Url/FacebookUrlDetectionHandler.php [hunk 1]
```diff
@@ -30,142 +30,139 @@
...
-    if ($protocol) {
-      return $this->protocolWithActiveSsl($protocol);
+    /**
+     * @inheritdoc
+     */
+    public function getCurrentUrl()
+    {
+        return $this->getHttpScheme() . '://'
+            . $this->getHostName()
+            . $this->getServerVar('REQUEST_URI');
```
Comment (yguedidi) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: Can be in one line IMO

File: src/Facebook/Url/FacebookUrlDetectionHandler.php [hunk 2]
```diff
@@ -30,142 +30,139 @@
...
+     */
+    protected function protocolWithActiveSsl($protocol)
+    {
+        $protocol = strtolower((string) $protocol);
+        return (
+            $protocol === 'on'
+            || $protocol === '1'
+            || $protocol === 'https'
+            || $protocol === 'ssl'
+        );
```
Comment (yguedidi) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: One line with `in_array(..., array(), true)`;

File: src/Facebook/Url/FacebookUrlManipulator.php [hunk 3]
```diff
@@ -29,138 +29,140 @@
...
+     * @param string|null $string
+     *
+     * @return string|null
+     */
+    public static function forceSlashPrefix($string)
+    {
+        if (!$string) {
+            return $string;
+        }
+        return strpos($string, '/') === 0 ? $string : '/' . $string;
```
Comment (yguedidi) [inline src/Facebook/Url/FacebookUrlManipulator.php]: Emtpy line above

File: src/Facebook/Url/FacebookUrlDetectionHandler.php (hunk 2 above)
Comment (stevenmaguire) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: Good idea! I was not considering this type of refactoring when working through some PSR2 formatting scenarios. Thanks for catching and suggesting.

Comment (yguedidi) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: You're welcome

File: src/Facebook/Url/FacebookUrlDetectionHandler.php [hunk 4]
```diff
@@ -30,142 +30,132 @@
...
+     * Detects an active SSL protocol value.
+     *
+     * @param string $protocol
+     *
+     * @return boolean
+     */
+    protected function protocolWithActiveSsl($protocol)
+    {
+        $protocol = strtolower((string) $protocol);
+        return in_array($protocol, array('on','1','https','ssl'), true);
```
Comment (yguedidi) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: You miss the empty line above `return` :)

Comment (yguedidi) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: And spaces after commas in `array()`

Comment (stevenmaguire) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: You're the best.

arooound

**Comment by yguedidi** [comment]
Please see #356

File: src/Facebook/Url/FacebookUrlDetectionHandler.php (hunk 4 above)
Comment (yguedidi) [inline src/Facebook/Url/FacebookUrlDetectionHandler.php]: Thanks :)

**Comment by gfosco** [comment]
@yguedidi This is also covered by 356?

**Comment by yguedidi** [comment]
@gfosco yes it is

**Pull request closed without merging**